sql/plan/project.go

```diff
@@ -46,8 +46,24 @@ func NewProject(expressions []sql.Expression, child sql.Node) *Project {
 // Schema implements the Node interface.
 func (p *Project) Schema() sql.Schema {
 	var s = make(sql.Schema, len(p.Projections))
-	for i, e := range p.Projections {
-		s[i] = transform.ExpressionToColumn(e, AliasSubqueryString(e))
+	childSch := p.Child.Schema()
+	for i, expr := range p.Projections {
+		s[i] = transform.ExpressionToColumn(expr, AliasSubqueryString(expr))
+		if alias, isAlias := expr.(*expression.Alias); isAlias {
+			if gf, isGf := alias.Child.(*expression.GetField); isGf {
+				idx := gf.Index() - 1 // not sure why these are off by 1
```

Comment:
These are column ids, not indexes. You maybe want to compare the `GetField.Id()` to an underlying `TableIdNode.Columns()`, not the schema strings